### PR TITLE
Fp-13 Sidebar Component

### PR DESCRIPTION
**Pull request by DiegoVillasenorVilla**
PR checklist

### Description

- I created an isolated Sidebar component this will be used later on in the application.

- The component had be test by react testing library

- The sidebar component has two props: ISidebarMenuItem and ISidebarMenuCard.

     - The first one is compose with the next properties: id, label, icon and url.
     - ISidebarMenuCard is compose with the next propesties: id, displayName, photoUrl

But this two props will could be adapted to the needs of the project

- The sidebar have two states, that are controled with a class with a boolean: collapsed an expanded

image

image

### Testing Checklist

- [] Test cases have been added/updated
- [] Tested locally in browser (at least one other than Chrome)
- [] All tests pass

### What JIRA Ticket(s) does this PR address?

- [FP-13](https://cd95.atlassian.net/browse/FP-13)

### CODE REVIEW

File: src/components/sidebar/ISiderbarMenuItem.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import {
```
Comment (caarlosdamian) [inline src/components/sidebar/ISiderbarMenuItem.ts]: Este archivo va en utils.

File: src/components/sidebar/SidebarMenuCardView.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+export function SidebarMenuCardView({
+  card,
+  isOpen,
+}: SidebarMenuCardViewProps) {
+  return (
+    <div className="SidebarMenuCardView">
+      <img className="profile" src={card.photoUrl} alt="User" width="100%" />
+      <div className={classNames('profileInfo', isOpen ? '' : 'collapsed')}>
+        <div className="name">
+          {' '}
```
Comment (caarlosdamian) [inline src/components/sidebar/SidebarMenuCardView.tsx]: Eliminar esto

File: src/utils/classes.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export function classNames(...args: string[]): string {
```
Comment (caarlosdamian) [inline src/utils/classes.ts:1]: Para que usas esto ??? no es necesario

File: src/components/sidebar/ISidebarMenuCard.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import { ISidebarMenuCard } from '../../interfaces/index';
+
+export const card: ISidebarMenuCard = {
```
Comment (caarlosdamian) [inline src/components/sidebar/ISidebarMenuCard.ts]: Esto va en Utils

File: src/components/sidebar/SidebarMenu.scss [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+.hamburgerbutton{
+    cursor: pointer;
+    border: none;
+    background-color: #8D91A0;
+    color: #FFFFFF;
+    padding: 10px;
+    width: 68px;
+}
+
+.Search-content-collapsed{
```
Comment (caarlosdamian) [inline src/components/sidebar/SidebarMenu.scss]: no usar mayuzculas en clases

File: src/components/sidebar/SidebarMenu.tsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+export function SidebarMenu({ items, card }: SidebarMenuProps) {
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+
+  function handleClick() {
+    setIsOpen(!isOpen);
+  }
+  return (
+    <div
+      data-testid="sidebar"
+      className={classNames('SidebarMenu', isOpen ? 'expanded' : 'collapsed')}
```
Comment (caarlosdamian) [inline src/components/sidebar/SidebarMenu.tsx]: className ={isOpen ? 'expanded' : 'collapsed'}

eliminar la funcion className no es necesaria

File: src/components/sidebar/SidebarMenuCardView.tsx [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+export function SidebarMenuCardView({
+  card,
+  isOpen,
+}: SidebarMenuCardViewProps) {
+  return (
+    <div className="SidebarMenuCardView">
+      <img className="profile" src={card.photoUrl} alt="User" width="100%" />
+      <div className={classNames('profileInfo', isOpen ? '' : 'collapsed')}>
```
Comment (caarlosdamian) [inline src/components/sidebar/SidebarMenuCardView.tsx]: Eliminar la fucion classNames 

className={`profileInfo ${isOpen && ' '  || 'collapsed'}

File: src/components/sidebar/SidebarMenuItemView.tsx [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+export function SidebarMenuItemView({
+  item,
+  isOpen,
+}: SidebarMenuItemViewProps) {
+  return (
+    <div className="SidebarMenuItemView">
+      <a href={item.url}>
+        <div className={classNames('ItemContent', isOpen ? '' : 'collapsed')}>
```
Comment (caarlosdamian) [inline src/components/sidebar/SidebarMenuItemView.tsx]: Eliminar la funcion classNames 
className={`ItemContent ${isOpen ? '':'collapsed'}

File: src/components/sidebar/SidebarPage.tsx [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+  VscSettingsGear,
+} from 'react-icons/vsc';
+
+import { ISidebarMenuCard, ISidebarMenuItem } from '../../interfaces/index';
+import { SidebarMenu } from './SidebarMenu';
+import imageProfile from '../../assets/img_cv.jpg';
+import './SidebarMenu.scss';
+
+export function SidebarPage() {
+  const items: ISidebarMenuItem[] = [
```
Comment (caarlosdamian) [inline src/components/sidebar/SidebarPage.tsx:19]: esta parte va en Utils

File: src/utils/classes.ts (hunk 3 above)
Comment (caarlosdamian) [reply src/utils/classes.ts:1]: Eliminar no es necesaria.

File: src/components/sidebar/SidebarMenu.tsx [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+
+interface SidebarMenuProps {
+  items: ISidebarMenuItem[];
+  card: ISidebarMenuCard;
+}
+
+export function SidebarMenu({ items, card }: SidebarMenuProps) {
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+
+  function handleClick() {
```
Comment (caarlosdamian) [inline src/components/sidebar/SidebarMenu.tsx]: user nomenclature 
const handleToggle = () =>{
setIsOpen(!isOpen)
}

File: src/components/sidebar/SidebarMenu.tsx [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+import { classNames } from '../../utils/classes';
+
+import './SidebarMenu.scss';
+
+interface SidebarMenuProps {
+  items: ISidebarMenuItem[];
+  card: ISidebarMenuCard;
+}
+
+export function SidebarMenu({ items, card }: SidebarMenuProps) {
```
Comment (caarlosdamian) [inline src/components/sidebar/SidebarMenu.tsx:15]: export const SidebarMenu = () =>{

}

File: src/components/DataGrid.tsx [hunk 12]
```diff
@@ -4,13 +4,17 @@ import { columns } from '../utils/tableColumns';
 import { useGetCovidDataQuery } from '../hooks/useApi';
 import { Table } from './table/Table';
 import { Irow } from '../interfaces';
+import { SidebarPage } from './sidebar/SidebarPage';
+import './sidebar/SidebarMenu.scss';
 
 export function DataGrid() {
```
Comment (caarlosdamian) [inline src/components/DataGrid.tsx:10]: export const DataGrid =()=>{}

File: src/components/sidebar/SidebarMenuCardView.tsx [hunk 13]
```diff
@@ -0,0 +1,25 @@
+import { ISidebarMenuCard } from '../../interfaces/index';
+import { classNames } from '../../utils/classes';
+import './SidebarMenu.scss';
+
+interface SidebarMenuCardViewProps {
+  card: ISidebarMenuCard;
+  isOpen: boolean;
+}
+
+export function SidebarMenuCardView({
```
Comment (caarlosdamian) [inline src/components/sidebar/SidebarMenuCardView.tsx:10]: export const SidebarMenuCardview = () => {}

File: src/components/sidebar/SidebarMenuItemView.tsx [hunk 14]
```diff
@@ -0,0 +1,26 @@
+import { ISidebarMenuItem } from '../../interfaces/index';
+import { classNames } from '../../utils/classes';
+import './SidebarMenu.scss';
+
+interface SidebarMenuItemViewProps {
+  item: ISidebarMenuItem;
+  isOpen: boolean;
+}
+
+export function SidebarMenuItemView({
```
Comment (caarlosdamian) [inline src/components/sidebar/SidebarMenuItemView.tsx:10]: export const SidebarMenuItemView = () => {}

File: src/components/sidebar/SidebarPage.tsx [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+  VscBell,
+  VscSettingsGear,
+} from 'react-icons/vsc';
+
+import { ISidebarMenuCard, ISidebarMenuItem } from '../../interfaces/index';
+import { SidebarMenu } from './SidebarMenu';
+import imageProfile from '../../assets/img_cv.jpg';
+import './SidebarMenu.scss';
+
+export function SidebarPage() {
```
Comment (caarlosdamian) [inline src/components/sidebar/SidebarPage.tsx:18]: export const SidebarPage = () =>{}

File: src/components/sidebar/SidebarPage.tsx [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+    { id: '2', label: 'Calendar', icon: VscCalendar, url: '/' },
+    { id: '3', label: 'Message', icon: VscComment, url: '/' },
+    { id: '4', label: 'Projects', icon: VscBriefcase, url: '/' },
+    { id: '5', label: 'Progress', icon: VscGraphLine, url: '/' },
+    { id: '6', label: 'Goals', icon: VscIssues, url: '/' },
+    { id: '7', label: 'Notifications', icon: VscBell, url: '/' },
+    { id: '8', label: 'Settings', icon: VscSettingsGear, url: '/' },
+  ];
+
+  const card: ISidebarMenuCard = {
```
Comment (caarlosdamian) [inline src/components/sidebar/SidebarPage.tsx:35]: Esto va en utils

File: src/components/sidebar/SidebarMenu.scss [hunk 17]
```diff
@@ -0,0 +1,182 @@
...
+    transition:  $property 0.2 ease-in-out;
+}
+
+
+.dataGrid-container{
+    display: flex;
+}
+
+.SidebarPage{
+    flex: 2;
```
Comment (caarlosdamian) [inline src/components/sidebar/SidebarMenu.scss]: Elimina el flex:2 maybe elimine el espacio que queda al momento de ser colapse

File: src/components/sidebar/SidebarMenu.scss [hunk 18]
```diff
@@ -0,0 +1,182 @@
...
+}
+
+
+.SidebarMenu{
```
Comment (caarlosdamian) [inline src/components/sidebar/SidebarMenu.scss]: no usar mayuzculas ni camelCase sidebar-menu

File: src/components/sidebar/SidebarMenu.scss [hunk 19]
```diff
@@ -0,0 +1,182 @@
...
+    &.expanded{
+       width: 300px;
+    }
+
+    &.collapsed{
+       width: 90px;
+    }
+}
+
+.menu-button{
```
Comment (caarlosdamian) [inline src/components/sidebar/SidebarMenu.scss]: sidebar__menu-button

File: src/components/sidebar/SidebarMenu.scss [hunk 20]
```diff
@@ -0,0 +1,182 @@
...
+    display: flex;
+    justify-content: flex-start;
+    margin-top: 10px;
+    margin-left: 10px;
+}
+
+.hamburgerbutton{
```
Comment (caarlosdamian) [inline src/components/sidebar/SidebarMenu.scss]: sidebar__hambuerger-button

File: src/components/sidebar/SidebarMenu.scss (hunk 5 above)
Comment (caarlosdamian) [inline src/components/sidebar/SidebarMenu.scss]: sidebar__search-content--collapsed

File: src/components/sidebar/SidebarMenu.scss [hunk 21]
```diff
@@ -0,0 +1,182 @@
...
+}
+
+.Search-content-collapsed{
+    display: flex;
+    justify-content: center;
+    margin-top: 10px;
+    margin-right: 30px;
+}
+
+.Search-content-expanded{
```
Comment (caarlosdamian) [inline src/components/sidebar/SidebarMenu.scss]: sidebar__content--expanded

File: src/components/sidebar/SidebarMenu.scss [hunk 22]
```diff
@@ -0,0 +1,182 @@
...
+    margin-top: 10px;
+    background: #F6F8FA;
+    border-radius: 5.54012px;
+    width: 278px;
+    height: 46px;
+    align-items: center;
+    margin-left: 10px;
+}
+
+.search-input{
```
Comment (caarlosdamian) [inline src/components/sidebar/SidebarMenu.scss]: sidebar__search-input

File: src/components/sidebar/SidebarMenu.scss [hunk 23]
```diff
@@ -0,0 +1,182 @@
...
+    display: flex;
+    flex-direction: row;
+    align-items: center;
+    overflow: hidden;
+    white-space: nowrap;
+    padding-left: 20px;
+}
+
+.search-icon{
```
Comment (caarlosdamian) [inline src/components/sidebar/SidebarMenu.scss]: sidebar__search-icon

File: src/components/sidebar/SidebarMenu.scss [hunk 24]
```diff
@@ -0,0 +1,182 @@
...
+    padding-left: 0px;
+}
+
+.input-search{
```
Comment (caarlosdamian) [inline src/components/sidebar/SidebarMenu.scss]: sidebar__input-search

File: src/components/sidebar/SidebarMenu.scss [hunk 25]
```diff
@@ -0,0 +1,182 @@
...
+    background: #F6F8FA;
+    border: none;
+    outline: none;
+    color: #8D91A0;
+    padding-left: 10px;
+}
+
+
+.SidebarMenuItemView{
```
Comment (caarlosdamian) [inline src/components/sidebar/SidebarMenu.scss]: sidebar__menu-item-view

File: src/components/sidebar/SidebarMenu.scss [hunk 26]
```diff
@@ -0,0 +1,182 @@
...
+        color: #8D91A0;
+        text-decoration: none;
+        display: block;
+        padding: 10px;
+        width: 100%;
+        border-radius: 5px;
+    }
+}
+
+.ItemContent{
```
Comment (caarlosdamian) [inline src/components/sidebar/SidebarMenu.scss]: sidebar__item-content

File: src/components/sidebar/SidebarMenu.scss [hunk 27]
```diff
@@ -0,0 +1,182 @@
...
+            display: inline-block;
+            animation-duration: 0.4s;
+            animation-name: tooltipe;
+        }
+    }
+
+}
+
+
+.label{
```
Comment (caarlosdamian) [inline src/components/sidebar/SidebarMenu.scss]: sidebar__label

File: src/components/sidebar/SidebarMenu.scss [hunk 28]
```diff
@@ -0,0 +1,182 @@
...
+    font-weight: 500px;
+    margin-left: 30px;
+    overflow: hidden;
+}
+
+
+.SidebarMenuCardView{
```
Comment (caarlosdamian) [inline src/components/sidebar/SidebarMenu.scss]: sidebar__menu-card-view

File: src/components/sidebar/SidebarMenu.scss [hunk 29]
```diff
@@ -0,0 +1,182 @@
...
+    display: flex;
+    align-items: center;
+    text-align: center;
+    flex-direction: column;
+
+}
+
+.profile{
```
Comment (caarlosdamian) [inline src/components/sidebar/SidebarMenu.scss]: sidebar__profile--img

File: src/components/sidebar/SidebarMenu.scss [hunk 30]
```diff
@@ -0,0 +1,182 @@
...
+    border-radius: 50px;
+    padding: 10px;
+    box-sizing: border-box;
+    width: 60px;
+    height: 60px;
+    
+}
+
+.profileInfo{
```
Comment (caarlosdamian) [inline src/components/sidebar/SidebarMenu.scss]: sidebar__profile-info

File: src/components/sidebar/SidebarMenu.scss [hunk 31]
```diff
@@ -0,0 +1,182 @@
...
+    height: 80px;
+    @include transition(height);
+
+    &.collapsed{
+        opacity: 0;
+        height: 0;
+    }
+}
+
+.name{
```
Comment (caarlosdamian) [inline src/components/sidebar/SidebarMenu.scss]: sidebar__name

**Review by caarlosdamian — CHANGES_REQUESTED** [review]
algunos comentarios

File: src/utils/classes.ts (hunk 3 above)
Comment (DiegoVillasenorVilla) [reply src/utils/classes.ts:1]: Lo usaba para la classname, para cuando era collapsed o se expandia

**Pull request closed without merging**